include/xtensor-io/xhighfive.hpp

```diff
@@ -0,0 +1,666 @@
+        * "/path" and "/path/to".
+        *
+        * @param file opened HighFive::File
+        * @param path path of the DataSet
+        */
+        inline void createGroup(HighFive::File &file, const std::string &path)
+        {
+            // find first "/"
+            std::size_t idx = path.find("/");
+
+            // loop over all groups
+            while (true)
+            {
+                // terminate if all "/" have been found
+                if (std::string::npos == idx)
+                {
+                    return;
+                }
+
+                // create group if needed
+                if (idx > 0)
+                {
+                    // get group name
+                    std::string name(path.substr(0,idx));
+
+                    // create if needed
+                    if (!file.exist(name))
+                    {
+                        file.createGroup(name);
+                    }
+                }
+
+                // proceed to next "/"
+                idx = path.find("/",idx+1);
+            }
+        }
+    }
+
+    namespace detail
+    {
+        inline auto highfive_file_mode(xt::file_mode mode)
+        {
+            switch (mode)
+            {
+                case xt::file_mode::create :
+                  return HighFive::File::Create;
+                case xt::file_mode::overwrite :
+                  return HighFive::File::Overwrite;
+                case xt::file_mode::append :
+                  return HighFive::File::ReadWrite;
+                case xt::file_mode::read :
+                  return HighFive::File::ReadOnly;
+                default :
+                  return HighFive::File::ReadOnly;
+            }
+        }
+
+        namespace scalar
+        {
+            template <class T, class E = void>
+            struct dump_impl
+            {
+                template <class C>
+                static HighFive::DataSet run(HighFive::File &file, const std::string &path, const C &data)
+                {
+                    extensions::createGroup(file, path);
+
+                    HighFive::DataSet dataset = file.createDataSet<C>(path, HighFive::DataSpace::From(data));
+
+                    dataset.write(data);
+
+                    file.flush();
+
+                    return dataset;
+                }
+            };
+
+            template <class T, class E = void>
+            struct overwrite_impl
+            {
+                template <class C>
+                static HighFive::DataSet run(HighFive::File &file, const std::string &path, const C &data)
+                {
+                    if (!extensions::exist(file, path))
+                    {
+                        return detail::scalar::dump_impl<T>::run(file, path, data);
+                    }
+
+                    HighFive::DataSet dataset = file.getDataSet(path);
+
+                    auto dataspace = dataset.getSpace();
+
```

Comment:
I don't feel all these empty lines are necessary, but up to you.

Reply:
Not sure, I will see what it looks like.